content/en/blog/_posts/2022-04-15-Increasing-the-security-bar-in-Ingress-NGINX/index.md

```diff
@@ -0,0 +1,127 @@
+---
+layout: blog
+title: 'Increasing the security bar in Ingress-NGINX'
+date: 2022-04-15
+slug: ingress-nginx-security-release
+---
+
+**Authors:** Ricardo Katz (VMware), James Strong (Chainguard)
+
+Ingress may be one of the most targeted components of Kubernetes. In fact, it usually is an HTTP Proxy, exposed to the Internet, containing multiple websites, and with some privileged access to Kubernetes API (mainly secret access for TLS certificates).
+
+While it is a risky component in your architecture, it is still required to properly expose your services.
+
+Ingress NGINX has been part of security assessments that figured out we have a big problem: we don't do all proper sanitization before turning the configuration into an nginx.conf file, which may lead to users being able to extract information.
+
+While we understand this risk and the real need to fix this, it's not an easy process to do, so we took another approach to reduce (but not remove!) this risk in the next release.
+
+## Meet Ingress NGINX v1.2.0 and the chrooted NGINX process
+
+One of the main problems is that we run our proxy server (NGINX) together with the controller (the component that has access to Kubernetes API Server and created the nginx.conf file). 
+
+So, NGINX does have the same access to the filesystem of the controller (and Kubernetes service account token, and other configurations from the container). While splitting those components is our end goal, we needed a fast response, that lead to the chroot idea.
+
+Let's take a look into what an Ingress Controller container looked like before this change:
+
+![Ingress NGINX pre chroot](ingress-pre-chroot.png)
+
+As we can see, the same container (not the Pod, the container!) that provides HTTP Proxy is the one that watches Ingress objects and writes the Container Volume
+
+Now, meet the new architecture:
+
+![Ingress NGINX post chroot](ingress-post-chroot.png)
+
+What does all of this mean? A good summary is: that we are running a new container inside our own container.
```

Comment:
```suggestion
What does all of this mean? A basic summary is: that we are isolating the NGINX webserver as a container inside the
controller container.
```

Reply:
s/webserver/proxy (or even isolating the NGINX Service). 

Will fix on another commit

Reply:
same, edited here